include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -101,12 +119,17 @@ class cuda_async_memory_resource final : public device_memory_resource {
     auto const pool_size = initial_pool_size.value_or(free / 2);
     auto* ptr            = do_allocate(pool_size, cuda_stream_default);
     do_deallocate(ptr, pool_size, cuda_stream_default);
-
+  }
 #else
+  // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+  cuda_async_memory_resource(thrust::optional<std::size_t> initial_pool_size             = {},
```

Comment:
Why did we move the macros to the outside of the function body? It requires us to repeat the function signature here. I though it was nice to avoid that redefinition.

Reply:
Oops. artifact of intermediate version where there were two different interfaces (which was ugly).

Reply:
Fixed.